optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/locator/EqualsLocationStrategy.java

```diff
@@ -22,7 +22,11 @@
 
     @Override
     public void addWorkingObject(Map<Object, Object> idToWorkingObjectMap, Object workingObject) {
-        idToWorkingObjectMap.put(workingObject, workingObject);
+        Object oldAddedObject = idToWorkingObjectMap.put(workingObject, workingObject);
+        if (oldAddedObject != null) {
+            throw new IllegalStateException("An object that is equal to workingObject (" + workingObject
+                    + ") has already been added.");
+        }
```

Comment:
So I added this check so the methods behaves similar to the one of PlanningIdLocationStrategy. If it turns out this check is to strict it can be removed anytime. If there are no objections, the PR can be merged as is.